client/components/select-dropdown/test/index.js

```diff
@@ -1,253 +1,215 @@
-	}
-
-	function getDropdownOptions() {
-		return [
-			{ value: 'status-options', label: 'Statuses', isLabel: true },
-			{ value: 'published', label: 'Published' },
-			{ value: 'scheduled', label: 'Scheduled' },
-			{ value: 'drafts', label: 'Drafts' },
-			null,
-			{ value: 'trashed', label: 'Trashed' },
-		];
-	}
-
-	function createKeyEvent( keyCode ) {
-		return {
-			keyCode,
-			preventDefault: jest.fn(),
-		};
-	}
+function getDropdownOptions() {
+	return [
+		{ value: 'status-options', label: 'Statuses', isLabel: true },
+		{ value: 'published', label: 'Published' },
+		{ value: 'scheduled', label: 'Scheduled' },
+		{ value: 'drafts', label: 'Drafts' },
+		null,
+		{ value: 'trashed', label: 'Trashed' },
+	];
+}
+
+function renderDropdown( props ) {
+	return render( <SelectDropdown options={ getDropdownOptions() } { ...props } /> );
+}
+
+describe( 'component rendering', () => {
+	test( 'should render a list with the provided options', () => {
+		renderDropdown();
+		expect( screen.getByRole( 'presentation' ) ).toHaveTextContent( 'Statuses' );
+		expect( screen.queryAllByRole( 'listitem' ) ).toHaveLength( 4 );
+	} );
+
+	test( 'should render a separator in place of any falsy option', () => {
+		renderDropdown();
+		expect( screen.queryAllByRole( 'separator' ) ).toHaveLength( 1 );
+	} );
+
+	test( 'should be initially closed', () => {
+		renderDropdown();
+		expect( screen.getByRole( 'button' ) ).toHaveAttribute( 'aria-expanded', 'false' );
+	} );
+
+	test( 'should execute toggleDropdown when clicked', async () => {
+		renderDropdown();
+
+		const btn = screen.getByRole( 'button' );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'false' );
+
+		await userEvent.click( btn );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'true' );
+
+		await userEvent.click( btn );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'false' );
+	} );
+
+	test( 'should not respond when clicked when disabled', async () => {
+		renderDropdown( { disabled: true } );
+
+		const btn = screen.getByRole( 'button' );
+		expect( btn ).toHaveAttribute( 'aria-disabled', 'true' );
+
+		await userEvent.click( btn );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'false' );
+	} );
+
+	test( 'should be possible to open the dropdown via keyboard', async () => {
+		renderDropdown();
+
+		const btn = screen.getByRole( 'button' );
+
+		await userEvent.tab();
+		expect( btn ).toHaveFocus();
+
+		await userEvent.keyboard( '[Space]' );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'true' );
+	} );
+} );
+
+describe( 'selected items', () => {
+	test( 'should return the initially selected value (if any)', () => {
+		renderDropdown( { initialSelected: 'drafts' } );
+
+		const selected = screen.queryByRole( 'menuitem', { current: true } );
+		expect( selected ).toHaveTextContent( 'Drafts' );
+
+		const btn = screen.getByRole( 'button' );
+		expect( btn.firstChild.textContent ).toBe( 'Drafts' );
```

Comment:
We'd get an [ESLint error](https://github.com/testing-library/eslint-plugin-jest-dom/blob/main/docs/rules/prefer-to-have-text-content.md) if we had the `jest-dom` ESLint rules enabled. It'd suggest that we use `toHaveTextContent()`.

```suggestion
		expect( btn.firstChild ).toHaveTextContent( 'Drafts' );
```

Reply:
Fixed via https://github.com/Automattic/wp-calypso/pull/63966/commits/9c8e2a32c218f6d14fa16af534505bcb5d65b3d0 👍🏻